docs/guide/custom-functions.md

```diff
@@ -119,72 +108,243 @@ Register your function plugin (and its translations) so that HyperFormula can re
 // prepare spreadsheet data
-const data = [['=HYPER()']];
+const data = [['Anthony', '=GREET(A1)']];
 
 // build a HyperFormula instance where you can use your function directly
 const hfInstance = HyperFormula.buildFromArray(data);
 
 // read the value of cell A1
-const result = hfInstance.getCellValue({ sheet: 0, col: 0, row: 0 });
+const result = hfInstance.getCellValue({ sheet: 0, col: 1, row: 0 });
 
 // open the browser's console to see the results
 console.log(result);
 ```
 
 ### Full example
 
-```javascript
-import { FunctionPlugin } from 'hyperformula';
+The complete implementation of this custom function is also included in the [demo](#working-demo).
 
-export class CountHF extends FunctionPlugin {
-  hyper(ast, state) {
-    return this.runFunction(ast.args, state, this.metadata('HYPER'), () => 'Hyperformula'.length);
+```js
+import { FunctionPlugin, CellError } from 'hyperformula';
+
+export class MyCustomPlugin extends FunctionPlugin {
+  greet(ast, state) {
+    return this.runFunction(
+      ast.args,
+      state,
+      this.metadata('GREET'),
+      (firstName) => {
+        return `👋 Hello, ${firstName}!`;
+      },
+    );
   }
 }
 
-CountHF.implementedFunctions = {
-  HYPER: {
-    method: 'hyper',
-  },
+MyCustomPlugin.implementedFunctions = {
+  GREET: {
+    method: 'greet',
+    parameters: [
+      { argumentType: 'STRING' }
+    ],
+  }
 };
 
-CountHF.translations = {
+MyCustomPlugin.translations = {
   enGB: {
-    HYPER: 'HYPER',
+    GREET: 'GREET',
   },
+  enUS: {
+    GREET: 'GREET',
+  }
+};
+
+HyperFormula.registerFunctionPlugin(MyCustomPlugin, MyCustomPlugin.translations);
+```
+
+## Advanced custom function example
+
+In a more advanced example, we'll create a custom function `DOUBLE_RANGE` that takes a range of numbers and returns the range of the same size with all the numbers doubled.
+
+### Accept a range argument
+
+To accept a range argument, declare it in the `parameters` array:
+
+```js
+MyCustomPlugin.implementedFunctions = {
+  DOUBLE_RANGE: {
+    method: 'doubleRange',
+    parameters: [
+      { argumentType: 'RANGE' },
+    ],
+  }
 };
+```
+
+The range arguments are passed to the implementation method as instances of the [`SimpleRangeValue` class](../api/classes/simplerangevalue.md):
+
+```js
+export class MyCustomPlugin extends FunctionPlugin {
+  doubleRange(ast, state) {
+    return this.runFunction(
+      ast.args,
+      state,
+      this.metadata('DOUBLE_RANGE'),
+      (range) => {
+        const rangeData = range.data;
+        // ...
+      },
+    );
+  }
+}
+```
+
+### Return an array of data
+
+A function may return multiple values in the form of an array. To do that, use [`SimpleRangeValue` class](../api/classes/simplerangevalue.md):
+
+```js
+export class MyCustomPlugin extends FunctionPlugin {
+  doubleRange(ast, state) {
+    return this.runFunction(
+      ast.args,
+      state,
+      this.metadata('DOUBLE_RANGE'),
+      (range) => {
+        const resultArray = //...
```

Comment:
Is it on purpose that the body of the function is omitted?

Reply:
Yes, this snippet demonstrates usage of `SimpleRangeValue` to return an array

Reply:
In my opinion it would unnecessarily increase the complexity of the guide. A reader can see the fully working example in the demo section if interested in the details.